### PR TITLE
Add allowArbitraryValues in OwnedEntityPicker

### PR DESCRIPTION
**Pull request by joanayma**
## Hey, I just made a Pull Request!

The `allowArbitraryValues` was added in Pull Request #8441 for `EntityPicker`. This feature does not exist in `OwnedEntityPicker`.
Also a comment from @GoFightNguyen was asking about this:
> @GoFightNguyen are we able to also add this to the new OwnedEntityPicker as well?
> 
> _Originally posted by @regicsolutions in https://github.com/backstage/backstage/issues/8441#issuecomment-997299196_

Based on the mentioned PR, I added this option to `OwnedEntityPicker`. I don't have a typescript or js background so anything wrong please let me know :slightly_smiling_face: 

The current behaviour can be check using the Template Editor:
image

The allowArbitraryValues set to false behaviour working with this patch:
image

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by benjdlambert** [comment]
Hey :wave: You'll also want to add this to the input props too under this type: https://github.com/backstage/backstage/blob/14146703e5bb5389316e006c30afadf8f7f5a4b9/plugins/scaffolder/src/components/fields/OwnedEntityPicker/OwnedEntityPicker.tsx#L39

File: .changeset/weak-jeans-cry.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-scaffolder': patch
+---
+
+Add allowArbitraryValues to ui:options in OwnedEntityPicker uiSchema, similar to allowArbitraryValues in EntityPicker
```
Comment (benjdlambert) [inline .changeset/weak-jeans-cry.md:5]:
```suggestion
Add `allowArbitraryValues` to `ui:options` in `OwnedEntityPicker`, similar to `allowArbitraryValues` in `EntityPicker`
```

**Review by benjdlambert — COMMENTED** [review]
Just one last nit I think

File: .changeset/weak-jeans-cry.md (hunk 1 above)
Comment (joanayma) [reply .changeset/weak-jeans-cry.md]: done!

**Review by benjdlambert — APPROVED** [review]

**Comment by joanayma** [comment]
@benjdlambert what failed on the CI?

**Pull request merged**